src/main/scala/com/fulcrumgenomics/umi/Umis.scala

```diff
@@ -80,8 +80,9 @@ object Umis {
       Some(name.substring(idx + 1, name.length))
     }
 
-    val umi   = rawUmi.map(raw => (if (raw.indexOf('+') > 0) raw.replace('+', '-') else raw).toUpperCase)
-    val valid = umi.forall(u => u.forall(isValidUmiCharacter))
+    val umiSeq = rawUmi.map(seq => (if (prefix.isEmpty) seq else seq.stripPrefix(prefix.get)))
```

Comment:
Although `.get()` exists, try to avoid it in Scala if possible since it relies on a guarantee that somewhere earlier (near or far away) there is a value contained in the option. That type of logical decoupling can introduce bugs/regressions in the future if a refactoring removes that assumption by accident.

```suggestion
    val umiSeq = rawUmi.flatMap(raw => prefix.map(raw.stripPrefix))
```